synapse/replication/tcp/resource.py

```diff
@@ -52,13 +42,18 @@ class ReplicationStreamProtocolFactory(Factory):
     """
 
     def __init__(self, hs):
-        self.streamer = hs.get_replication_streamer()
+        self.handler = hs.get_tcp_replication()
         self.clock = hs.get_clock()
         self.server_name = hs.config.server_name
+        self.hs = hs
+
+        # Ensure the replication streamer is started if we register a
```

Comment:
Well, you'd always do something like `listen_tcp("localhost", 80, ReplicationStreamProtocolFactory(hs))` or whatever. Its not a fantastic place to put it, but not sure where else would be a good place

Reply:
The subtlety here is that we only really want to start this if we configure a replication listener. I guess we could put this in `HomeServer.setup` and then check if we have a replication listener? But not sure that is a better place.

Reply:
ok so what you're saying is:

> The fact that we have got here implies that we are trying to register a replication server endpoint, so we should make sure that we start the replication streamer so that we have something to send to the clients when they connect.
>
> This isn't terribly elegant, but we don't really want to start a replication streamer *unless* we have a replication server endpoint, which makes it awkward to put this call anywhere else.

which is fine, but please can you make the comment say so? :)